File: ecom-frontend/src/app/models/Order.ts
import {ProductItem} from "./ProductItem";

enum OrderStatus {
  PENDING="PENDING",
  COMPLETED="COMPLETED",
  CANCELLED="CANCELLED",
  SHIPPED="SHIPPED",
  DELIVERED="DELIVERED",
}

export interface Order {
  id: number;
  createdAt: string;
  total: number;
  status: OrderStatus;
  productItems: ProductItem[];
}

File: billing-service/src/main/java/ma/enset/billingservice/entities/Order.java
package ma.enset.billingservice.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import ma.enset.billingservice.enums.OrderStatus;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Builder
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDateTime createdAt;
    @Enumerated(EnumType.STRING)
    private OrderStatus status;
    @OneToMany(mappedBy = "order")
    private List<ProductItem> productItems;
}